src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/PSUserAgent.cs

```diff
@@ -118,19 +119,39 @@ internal static string App
             }
         }
 
-        internal static string Platform
+        internal static string PlatformName
         {
             get
             {
-                return ("Windows NT");
+                if (Platform.IsWindows)
+                {
+                    // find the version in the windows operating system description
+                    string versionText = PSUserAgent.OS.Substring(PSUserAgent.OS.TrimEnd().LastIndexOf(" ") +1);
+                    Version windowsPlatformversion = new Version(versionText);
```

Comment:
Since this would be dynamically generated each time the Web Cmdlets are called, but is static beyond the initial call, perhaps adding a private field `_windowsPlatformversionString`  (or something) generated once on the first call (`if (_windowsPlatformversionString == null)`) and then reference it directly every time thereafter. `return  _windowsPlatformversionString`

Reply:
It is good candidate for private static variable and private static initialization function.